DiscountDiscover/AppDelegate.m

```diff
@@ -36,9 +40,52 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     // Google Maps
     [GMSServices provideAPIKey:[APIManager getAPIKey:@"GoogleAPIKey"]];
     
+    // notification authorization
+    UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+    [center requestAuthorizationWithOptions:(UNAuthorizationOptionBadge | UNAuthorizationOptionSound | UNAuthorizationOptionAlert) completionHandler:^(BOOL granted, NSError * _Nullable error) {}];
+    
+    // background fetch frequency restriction
+    const int minimumTime = 10800; // 3 hours
+    [application setMinimumBackgroundFetchInterval:minimumTime];
+    
+    [UNUserNotificationCenter currentNotificationCenter].delegate = self;
+    
     return YES;
 }
 
+-(void)application:(UIApplication *)application performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler{
+    
+    APIManager *manager = [[APIManager alloc] init];
+    [manager fetchNearbyDeal:^(Deal *deal, UIBackgroundFetchResult result) {
+        if (deal != nil) {
+            [self scheduleNotification:deal];
+        }
+        completionHandler(result);
+    }];
+}
+
+-(void)scheduleNotification:(Deal *)deal {
+    
+    // configure notification content
```

Comment:
All the the notifications logic shouldn't be on the `AppDelegate` directly, it should be encapsulated in a dedicated class like `DealNotificationsManager` or similar.